app/controllers/devise_registration_controller.rb

```diff
@@ -23,39 +23,78 @@ class DeviseRegistrationController < Devise::RegistrationsController
   def start
     render :closed and return unless Rails.configuration.enable_registration
 
-    redirect_to url_for_state and return unless registration_state.state == "start"
+    payload = get_payload(params[:jwt])
+
+    @email = params.dig(:user, :email)
+    @password = params.dig(:user, :password) # pragma: allowlist secret
+    @phone = params.dig(:user, :phone)
+    @criteria_keys = payload.dig(:attributes, :transition_checker_state, "criteria_keys") if payload
 
     return if request.get?
 
-    password = params.dig(:user, :password) # pragma: allowlist secret
-    password_confirmation = params.dig(:user, :password_confirmation)
-    password_length_ok = Devise.password_length.include? password&.length
-    password_confirmation_ok = password == password_confirmation
+    password_length_ok = Devise.password_length.include? @password&.length
```

Comment:
Since this is only used in one place now, I think it would be better to just move this condition into the `elsif` directly